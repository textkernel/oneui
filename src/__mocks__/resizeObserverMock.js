export class ResizeObserverMock {
    observe = jest.fn();

    unobserve = jest.fn();

    disconnect = jest.fn();
}
