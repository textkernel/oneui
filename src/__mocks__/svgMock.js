/* eslint-disable react/display-name, react/destructuring-assignment */
import React from 'react';

module.exports = (props) => <svg data-testid={props['data-testid'] || 'default-icon'} />;
