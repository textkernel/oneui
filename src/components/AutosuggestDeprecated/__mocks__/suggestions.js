export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptune' },
    { name: 'Pluto', disabled: true },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
