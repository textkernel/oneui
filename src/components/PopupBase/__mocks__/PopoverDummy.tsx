import * as React from 'react';
import { Button } from '../../..';

interface Props {
    setPopupVisibility: (isVisible) => void;
}
export const PopoverDummy = React.forwardRef<HTMLDivElement, Props>(
    ({ setPopupVisibility }, ref) => {
        const popupStyle = {
            backgroundColor: '#FFEEDD',
            padding: '10px',
            borderRadius: '5px',
            border: '1px solid #CCC',
            boxShadow: '0 3px 6px 6px rgba(0, 0, 0, 0.05)',
        };
        return (
            <div style={popupStyle} ref={ref} role="group">
                <p>I am the content of pop up.</p>
                <p>Can be a form or anything.</p>
                <Button onClick={() => setPopupVisibility(false)}>Close me!</Button>
            </div>
        );
    }
);

PopoverDummy.displayName = 'Popover';
