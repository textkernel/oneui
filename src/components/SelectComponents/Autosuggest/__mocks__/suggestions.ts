export const SUGGESTIONS = [
    { name: 'Sun' },
    { name: 'Moon' },
    { name: 'Mercury' },
    { name: 'Venus' },
    { name: 'Earth' },
    { name: 'Mars' },
    { name: 'Jupiter' },
    { name: 'Saturn' },
    { name: 'Neptune' },
    { name: 'Pluto', disabled: true },
];

export type SuggestionsType = (typeof SUGGESTIONS)[number];

export const COMPLEX_SUGGESTIONS = [
    { name: 'Sun', type: 'star' },
    { name: 'Moon', type: 'star' },
    { name: 'Mercury', type: 'planet' },
    { name: 'Venus', type: 'planet' },
    { name: 'Earth', type: 'planet' },
    { name: 'Mars', type: 'planet' },
    { name: 'Jupiter', type: 'planet' },
    { name: 'Saturn', type: 'planet' },
    { name: 'Neptune', type: 'planet' },
];

export const SUGGESTION_TO_STRING = (item) => (item ? item.name : '');
