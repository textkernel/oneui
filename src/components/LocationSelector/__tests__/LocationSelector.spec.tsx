import React from 'react';
import { render, screen, RenderResult, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { LocationSelector, LocationSelectorLocation } from '..';
import { stabGoogleApi } from '../../../__mocks__/googleApiMock';

stabGoogleApi();

describe('LocationSelector component', () => {
    const selectedLocations = [
        {
            id: 'ajdo-219a-j19v-0491',
            center: {
                lng: 4.894539799999961,
                lat: 52.3666969,
            },
        } as LocationSelectorLocation,
        {
            id: 'ajdo-219a-j19v-0492',
            center: {
                lng: 5.121420100000023,
                lat: 52.09073739999999,
            },
        } as LocationSelectorLocation,
    ];

    const minRadius = 1;
    const maxRadius = 100;
    const radiusStep = 1;
    const radiusDefaultValue = 2;
    const renderRadiusLabel = (r) => `+ ${r} km`;
    const country = 'NL';
    const placeTypes = ['(regions)'];

    const inputPlaceholder = 'inputPlaceholder';
    const noSuggestionsPlaceholder = 'noSuggestionsPlaceholder';
    const doneLabel = 'doneLabel';

    const onAddLocationMock = jest.fn();
    const onUpdateLocationMock = jest.fn();
    const onRemoveLocationMock = jest.fn();
    const onLocationAutocompleteErrorMock = jest.fn();

    let view: RenderResult;

    beforeEach(() => {
        view = render(
            <LocationSelector
                apiKey="apiKey"
                language="en"
                radiusUnits="km"
                radiusDefaultValue={radiusDefaultValue}
                modalContentLabel="Location selection dialog"
                clearTooltipLabel="Clear"
                inputPlaceholder={inputPlaceholder}
                hasRadius
                minRadius={minRadius}
                maxRadius={maxRadius}
                radiusStep={radiusStep}
                renderRadiusLabel={renderRadiusLabel}
                onRemoveLocation={onRemoveLocationMock}
                doneLabel={doneLabel}
                country={country}
                placeTypes={placeTypes}
                noSuggestionsPlaceholder={noSuggestionsPlaceholder}
                showCountryInSuggestions
                onLocationAutocompleteError={onLocationAutocompleteErrorMock}
                onUpdateLocation={onUpdateLocationMock}
                selectedLocations={selectedLocations}
                onAddLocation={onAddLocationMock}
                onRemoveAllLocations={jest.fn()}
            />
        );
    });

    it('should render LocationSelector propertly when modal is closed', () => {
        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('img')).toBeInTheDocument();
        expect(screen.getAllByRole('button')).toHaveLength(2);
    });

    it('should render LocationSelector properly when modal is open', async () => {
        const user = userEvent.setup();
        await user.click(screen.getByRole('button', { name: 'inputPlaceholder' }));

        expect(view.container).toMatchSnapshot();
        expect(screen.getByRole('img')).toBeInTheDocument();
        expect(screen.getAllByRole('button')).toHaveLength(2);
        expect(screen.getByRole('status')).toBeInTheDocument();
    });

    it('should open modal when requested', async () => {
        const user = userEvent.setup();
        const button = screen.getByRole('button', { name: 'inputPlaceholder' });

        expect(button).not.toHaveAttribute('isOpen');

        await user.click(button);

        expect(screen.getByRole('dialog')).toBeVisible();

        await waitFor(() => {
            expect(button).not.toHaveAttribute('onRequestClose');
        });

        expect(screen.getByRole('dialog')).toBeVisible();
    });
});
