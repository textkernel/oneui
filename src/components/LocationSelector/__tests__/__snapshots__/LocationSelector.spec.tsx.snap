// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSelector component should render LocationSelector properly when modal is open 1`] = `
<div>
  <div>
    <div
      class="FieldWrapper FieldWrapper--isFocused LocationSelector__mainTextButtonWrapper"
      title="inputPlaceholder"
    >
      <div
        class="FieldWrapper__content"
      >
        <svg
          class="LocationSelector__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          />
        </svg>
        <button
          class="LocationSelector__mainTextButton LocationSelector__mainTextButton--muted"
          type="button"
        >
          inputPlaceholder
        </button>
      </div>
      <button
        class="Button Button--isLink Button--isInline FieldWrapper__clearButton"
        title="Clear"
        type="button"
      >
        <svg
          aria-label="Clear"
          fill="currentColor"
          height="16"
          role="img"
          size="16"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`LocationSelector component should render LocationSelector propertly when modal is closed 1`] = `
<div>
  <div>
    <div
      class="FieldWrapper LocationSelector__mainTextButtonWrapper"
      title="inputPlaceholder"
    >
      <div
        class="FieldWrapper__content"
      >
        <svg
          class="LocationSelector__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
          />
        </svg>
        <button
          class="LocationSelector__mainTextButton LocationSelector__mainTextButton--muted"
          type="button"
        >
          inputPlaceholder
        </button>
      </div>
      <button
        class="Button Button--isLink Button--isInline FieldWrapper__clearButton"
        title="Clear"
        type="button"
      >
        <svg
          aria-label="Clear"
          fill="currentColor"
          height="16"
          role="img"
          size="16"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
